{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compile Probe Annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Got to do this in rpy2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "HDFS_DIR = '/cellar/users/agross/Data/SSD/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from pandas.rpy.common import convert_robj\n",
    "import rpy2.robjects as robjects\n",
    "from pandas.rpy.common import convert_to_r_dataframe, convert_robj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Loading required package: IlluminaHumanMethylation450kanno.ilmn12.hg19\n",
       "Loading required package: minfi\n",
       "Loading required package: BiocGenerics\n",
       "Loading required package: parallel\n",
       "\n",
       "Attaching package: ‘BiocGenerics’\n",
       "\n",
       "The following objects are masked from ‘package:parallel’:\n",
       "\n",
       "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
       "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
       "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
       "\n",
       "The following object is masked from ‘package:stats’:\n",
       "\n",
       "    xtabs\n",
       "\n",
       "The following objects are masked from ‘package:base’:\n",
       "\n",
       "    anyDuplicated, append, as.data.frame, as.vector, cbind, colnames,\n",
       "    duplicated, eval, evalq, Filter, Find, get, intersect, is.unsorted,\n",
       "    lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
       "    pmin, pmin.int, Position, rank, rbind, Reduce, rep.int, rownames,\n",
       "    sapply, setdiff, sort, table, tapply, union, unique, unlist\n",
       "\n",
       "Loading required package: Biobase\n",
       "Welcome to Bioconductor\n",
       "\n",
       "    Vignettes contain introductory material; view with\n",
       "    'browseVignettes()'. To cite Bioconductor, see\n",
       "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
       "\n",
       "Loading required package: lattice\n",
       "Loading required package: reshape\n",
       "Loading required package: GenomicRanges\n",
       "Loading required package: IRanges\n",
       "\n",
       "Attaching package: ‘IRanges’\n",
       "\n",
       "The following objects are masked from ‘package:reshape’:\n",
       "\n",
       "    expand, rename\n",
       "\n",
       "Loading required package: XVector\n",
       "Loading required package: Biostrings\n",
       "Loading required package: bumphunter\n",
       "Loading required package: foreach\n",
       "foreach: simple, scalable parallel programming from Revolution Analytics\n",
       "Use Revolution R for scalability, fault tolerance and more.\n",
       "http://www.revolutionanalytics.com\n",
       "Loading required package: iterators\n",
       "Loading required package: locfit\n",
       "locfit 1.5-9.1 \t 2013-03-22\n",
       "\n",
       "Attaching package: ‘locfit’\n",
       "\n",
       "The following objects are masked from ‘package:GenomicRanges’:\n",
       "\n",
       "    left, right\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "require('IlluminaHumanMethylation450kanno.ilmn12.hg19')\n",
    "ann = IlluminaHumanMethylation450kanno.ilmn12.hg19\n",
    "data = ann@data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "islands = convert_robj(robjects.r(\"data.frame(data[['Islands.UCSC']])\"))\n",
    "locations = convert_robj(robjects.r(\"data.frame(data[['Locations']])\"))\n",
    "other = convert_robj(robjects.r(\"data.frame(data[['Other']])\"))\n",
    "other = other.ix[:, 2:] #first two columns are probe sequences which take up a lot of space\n",
    "snps = convert_robj(robjects.r(\"data.frame(data[['SNPs.Illumina']])\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "PATH = '/cellar/users/agross/Data/GeneSets/methylation450_annotations.csv'\n",
    "probe_annotations = pd.read_csv(PATH, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store = pd.HDFStore(HDFS_DIR + 'methylation_annotation.h5')\n",
    "store.append('islands', islands)\n",
    "store.create_table_index('islands', optlevel=9, kind='full')\n",
    "store.append('locations', locations)\n",
    "store.create_table_index('locations', optlevel=9, kind='full')\n",
    "store.append('other', other)\n",
    "store.create_table_index('other', optlevel=9, kind='full')\n",
    "store.append('snps', snps)\n",
    "store.create_table_index('snps', optlevel=9, kind='full')\n",
    "store.append('probe_annotations', probe_annotations)\n",
    "store.create_table_index('probe_annotations', optlevel=9, kind='full')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read in Flow Sorted Data and Convert to More Usable Format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Loading required package: minfiData\n",
       "Loading required package: IlluminaHumanMethylation450kmanifest\n",
       "Loading required package: doParallel\n",
       "Loading required package: FlowSorted.Blood.450k\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "require(minfi)\n",
    "require(minfiData)\n",
    "require(doParallel)\n",
    "\n",
    "referencePkg <- \"FlowSorted.Blood.450k\"\n",
    "require(referencePkg, character.only = TRUE)\n",
    "data(list = referencePkg)\n",
    "referenceRGset <- get(referencePkg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[preprocessQuantile] Mapping to genome.\n",
       "[preprocessQuantile] Fixing outliers.\n",
       "[preprocessQuantile] Quantile normalizing.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "Mset.quantile = preprocessQuantile(referenceRGset, removeBadSamples = TRUE)\n",
    "beta = getBeta(Mset.quantile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "beta = convert_robj(robjects.r(\"data.frame(beta)\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "labels = convert_robj(robjects.r('referenceRGset$CellType'))\n",
    "label_map = pd.Series(labels, beta.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store['label_map'] = label_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "beta.to_hdf(store,'flow_sorted_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "store.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: dynamicTreeCut\n",
      "Loading required package: flashClust\n",
      "\n",
      "Attaching package: ‘flashClust’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    hclust\n",
      "\n",
      "==========================================================================\n",
      "*\n",
      "*  Package WGCNA 1.41.1 loaded.\n",
      "*\n",
      "*    Important note: It appears that your system supports multi-threading,\n",
      "*    but it is not enabled within WGCNA in R. \n",
      "*    To allow multi-threading within WGCNA with all available cores, use \n",
      "*\n",
      "*          allowWGCNAThreads()\n",
      "*\n",
      "*    within R. Use disableWGCNAThreads() to disable threading if necessary.\n",
      "*    Alternatively, set the following environment variable on your system:\n",
      "*\n",
      "*          ALLOW_WGCNA_THREADS=<number_of_processors>\n",
      "*\n",
      "*    for example \n",
      "*\n",
      "*          ALLOW_WGCNA_THREADS=8\n",
      "*\n",
      "*    To set the environment variable in linux bash shell, type \n",
      "*\n",
      "*           export ALLOW_WGCNA_THREADS=8\n",
      "*\n",
      "*     before running R. Other operating systems or shells will\n",
      "*     have a similar command to achieve the same aim.\n",
      "*\n",
      "==========================================================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘WGCNA’\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    cor\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    cor\n",
      "\n",
      "Loading required package: RPMM\n",
      "Loading required package: cluster\n"
     ]
    }
   ],
   "source": [
    "robjects.r.library('WGCNA');\n",
    "robjects.r.source(\"/cellar/users/agross/Data/MethylationAge/Horvath/NORMALIZATION.R\");"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gold_standard = pd.read_csv('/cellar/users/agross/Data/MethylationAge/Horvath/probeAnnotation21kdatMethUsed.csv', \n",
    "                            index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def normalize_meth(df):\n",
    "    df = df.ix[gold_standard.index]\n",
    "    df = df.T.fillna(gold_standard.goldstandard2).T\n",
    "\n",
    "    df_r = robjects.r.t(convert_to_r_dataframe(df))\n",
    "    gs = list(gold_standard.goldstandard2.ix[df.index])\n",
    "    gs_r = robjects.FloatVector(gs)\n",
    "    data_n_r = robjects.r.BMIQcalibration(df_r, gs_r)\n",
    "    data_n = convert_robj(data_n_r).T\n",
    "    return data_n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Fitting EM beta mixture to goldstandard probes\"\n",
      "[1] Inf\n",
      "[1] 0.008547826\n",
      "[1] 0.007005981\n",
      "[1] 0.009442499\n",
      "[1] 0.01056228\n",
      "[1] \"Done\"\n",
      "ii= 1\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01023027\n",
      "[1] 0.009826291\n",
      "[1] 0.009266958\n",
      "[1] 0.008553946\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 2\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009436482\n",
      "[1] 0.009010343\n",
      "[1] 0.008323679\n",
      "[1] 0.007523282\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 3\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01068228\n",
      "[1] 0.01005551\n",
      "[1] 0.009282203\n",
      "[1] 0.008314943\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 4\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009733978\n",
      "[1] 0.009811863\n",
      "[1] 0.00962571\n",
      "[1] 0.009045181\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 5\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009489052\n",
      "[1] 0.009409222\n",
      "[1] 0.008959802\n",
      "[1] 0.008283887\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 6\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009324937\n",
      "[1] 0.009161499\n",
      "[1] 0.008681662\n",
      "[1] 0.007952304\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 7\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01075274\n",
      "[1] 0.009485142\n",
      "[1] 0.008580555\n",
      "[1] 0.007703105\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 8\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01047105\n",
      "[1] 0.009220373\n",
      "[1] 0.00818817\n",
      "[1] 0.007161418\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 9\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.0113729\n",
      "[1] 0.01018567\n",
      "[1] 0.009027107\n",
      "[1] 0.007945862\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 10\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01092163\n",
      "[1] 0.01100352\n",
      "[1] 0.01053349\n",
      "[1] 0.009762516\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 11\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01168212\n",
      "[1] 0.01128838\n",
      "[1] 0.01053688\n",
      "[1] 0.009547311\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 12\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01132091\n",
      "[1] 0.01113542\n",
      "[1] 0.01039278\n",
      "[1] 0.009354893\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 13\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01020447\n",
      "[1] 0.009712368\n",
      "[1] 0.009231272\n",
      "[1] 0.008571845\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 14\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009610878\n",
      "[1] 0.009545053\n",
      "[1] 0.009228907\n",
      "[1] 0.008651666\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 15\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.0100768\n",
      "[1] 0.009967384\n",
      "[1] 0.009595477\n",
      "[1] 0.008919797\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 16\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01247362\n",
      "[1] 0.01178717\n",
      "[1] 0.01106416\n",
      "[1] 0.01022293\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 17\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01161763\n",
      "[1] 0.01072436\n",
      "[1] 0.009896994\n",
      "[1] 0.008976952\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 18\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01317808\n",
      "[1] 0.01221463\n",
      "[1] 0.01129939\n",
      "[1] 0.01026172\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 19\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.0105017\n",
      "[1] 0.00920218\n",
      "[1] 0.008178436\n",
      "[1] 0.007170861\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 20\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01092858\n",
      "[1] 0.009448877\n",
      "[1] 0.008215955\n",
      "[1] 0.007062468\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 21\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01073185\n",
      "[1] 0.009432575\n",
      "[1] 0.008230508\n",
      "[1] 0.007120087\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 22\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01127176\n",
      "[1] 0.01071555\n",
      "[1] 0.01012174\n",
      "[1] 0.009302703\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 23\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.00970992\n",
      "[1] 0.009528439\n",
      "[1] 0.00917507\n",
      "[1] 0.008521863\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 24\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01073021\n",
      "[1] 0.01057176\n",
      "[1] 0.01012315\n",
      "[1] 0.009402837\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 25\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01039622\n",
      "[1] 0.009810437\n",
      "[1] 0.009078659\n",
      "[1] 0.008133983\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 26\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01182434\n",
      "[1] 0.0107445\n",
      "[1] 0.009637753\n",
      "[1] 0.008507685\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 27\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01080689\n",
      "[1] 0.01018908\n",
      "[1] 0.009363736\n",
      "[1] 0.008401483\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 28\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009868283\n",
      "[1] 0.009657039\n",
      "[1] 0.009146975\n",
      "[1] 0.008362317\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 29\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009605004\n",
      "[1] 0.009230724\n",
      "[1] 0.00856012\n",
      "[1] 0.00776864\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 30\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009988319\n",
      "[1] 0.009912126\n",
      "[1] 0.009402724\n",
      "[1] 0.00864858\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 31\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01083974\n",
      "[1] 0.009729307\n",
      "[1] 0.008578559\n",
      "[1] 0.007478666\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 32\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01006624\n",
      "[1] 0.009053035\n",
      "[1] 0.008061232\n",
      "[1] 0.007115429\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 33\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01054159\n",
      "[1] 0.00945053\n",
      "[1] 0.008468364\n",
      "[1] 0.007496301\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 34\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01056997\n",
      "[1] 0.01049988\n",
      "[1] 0.009923185\n",
      "[1] 0.009035759\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 35\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01217659\n",
      "[1] 0.01177213\n",
      "[1] 0.01101787\n",
      "[1] 0.01002794\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 36\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01171168\n",
      "[1] 0.01138887\n",
      "[1] 0.01074079\n",
      "[1] 0.009805231\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 37\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01078004\n",
      "[1] 0.01057884\n",
      "[1] 0.01011943\n",
      "[1] 0.009335859\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 38\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01128077\n",
      "[1] 0.01110618\n",
      "[1] 0.01048798\n",
      "[1] 0.009592993\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 39\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009769312\n",
      "[1] 0.009578483\n",
      "[1] 0.00922093\n",
      "[1] 0.008591407\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 40\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01246212\n",
      "[1] 0.01175322\n",
      "[1] 0.01088872\n",
      "[1] 0.009858786\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 41\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01225856\n",
      "[1] 0.01145393\n",
      "[1] 0.0106402\n",
      "[1] 0.00969996\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 42\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01191813\n",
      "[1] 0.01124039\n",
      "[1] 0.01042536\n",
      "[1] 0.009416451\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 43\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01070761\n",
      "[1] 0.009787715\n",
      "[1] 0.008684915\n",
      "[1] 0.007565564\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 44\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01033669\n",
      "[1] 0.009388771\n",
      "[1] 0.008262022\n",
      "[1] 0.00713916\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 45\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01125671\n",
      "[1] 0.01000996\n",
      "[1] 0.008767284\n",
      "[1] 0.007568184\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 46\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01113371\n",
      "[1] 0.0109196\n",
      "[1] 0.01039837\n",
      "[1] 0.009539701\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 47\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01079374\n",
      "[1] 0.01069245\n",
      "[1] 0.01001714\n",
      "[1] 0.009154231\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 48\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01071208\n",
      "[1] 0.01074448\n",
      "[1] 0.01024947\n",
      "[1] 0.009472749\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 49\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009843241\n",
      "[1] 0.009000582\n",
      "[1] 0.008035213\n",
      "[1] 0.00710348\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 50\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.0108197\n",
      "[1] 0.009909336\n",
      "[1] 0.008762327\n",
      "[1] 0.007660298\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 51\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01026029\n",
      "[1] 0.009330855\n",
      "[1] 0.008190156\n",
      "[1] 0.007074137\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 52\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01034261\n",
      "[1] 0.009491481\n",
      "[1] 0.008380126\n",
      "[1] 0.007251519\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 53\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01107918\n",
      "[1] 0.01008208\n",
      "[1] 0.009001733\n",
      "[1] 0.007909088\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 54\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01298345\n",
      "[1] 0.01190344\n",
      "[1] 0.01061476\n",
      "[1] 0.009380944\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 55\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009676665\n",
      "[1] 0.008861619\n",
      "[1] 0.007929816\n",
      "[1] 0.006968677\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 56\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01009102\n",
      "[1] 0.009093187\n",
      "[1] 0.007956899\n",
      "[1] 0.006865738\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 57\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.00976811\n",
      "[1] 0.008747122\n",
      "[1] 0.007736342\n",
      "[1] 0.006701171\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 58\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01022828\n",
      "[1] 0.009383221\n",
      "[1] 0.008293084\n",
      "[1] 0.007167398\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 59\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.009639483\n",
      "[1] 0.008778499\n",
      "[1] 0.007711985\n",
      "[1] 0.006644498\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n",
      "ii= 60\n",
      "[1] \"Fitting EM beta mixture to input probes\"\n",
      "[1] Inf\n",
      "[1] 0.01000023\n",
      "[1] 0.009147731\n",
      "[1] 0.008146869\n",
      "[1] 0.007070174\n",
      "[1] \"Done\"\n",
      "[1] \"Start normalising input probes\"\n"
     ]
    }
   ],
   "source": [
    "data_n = normalize_meth(beta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "data_n.to_hdf(HDFS_DIR, 'methylation_annotation.h5',\n",
    "              'flow_sorted_data_horvath_norm')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
